lib/ui/hash_codes.dart

```diff
@@ -2,122 +2,33 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-
 // @dart = 2.12
 part of dart.ui;
 
-class _HashEnd { const _HashEnd(); }
-const _HashEnd _hashEnd = _HashEnd();
-
-/// Jenkins hash function, optimized for small integers.
-//
-// Borrowed from the dart sdk: sdk/lib/math/jenkins_smi_hash.dart.
-class _Jenkins {
-  static int combine(int hash, Object? o) {
-    assert(o is! Iterable);
```

Comment:
How important is this assertion?

`Object.hash` has no similar assertion, and never will, since t is possible to have an `Iterable` that does define a well-behaved `hashCode`/`==`, e.g. `BuiltList` https://pub.dev/documentation/built_collection/latest/built_collection/BuiltList/hashCode.html)

Reply:
cc @Hixie @sigurdm

Reply:
I don't think it is very important - I think it is guarding against thinking you are doing a deep hash while you most probably are not (the default List has identity-based hashCode). This is not foolproof, and for example won't guard against Maps that have the same issue.

I agree that if an `Iterable` implements a non-identity hashCode then the assertion is not-needed.

Not sure if there is a way of asking that question.